File: client/android/UseCase/Auth/Session/ConfirmVerificationUseCase.kt
package app.mobiling.client.usecase.auth.session

import app.mobiling.client.contract.auth.session.AuthSessionPayload
import app.mobiling.client.contract.auth.session.ConfirmVerificationRequest
import app.mobiling.client.data.auth.session.AuthSessionGateway

class ConfirmVerificationUseCase(
    private val gateway: AuthSessionGateway,
) {
    suspend operator fun invoke(request: ConfirmVerificationRequest): AuthSessionPayload =
        gateway.confirmVerification(request)
}

File: client/ios/UseCase/Auth/Session/ConfirmVerificationUseCase.swift
public struct ConfirmVerificationUseCase {
    private let gateway: AuthSessionGateway

    public init(gateway: AuthSessionGateway) {
        self.gateway = gateway
    }

    public func callAsFunction(request: ConfirmVerificationRequest) async throws -> AuthSessionPayload {
        try await gateway.confirmVerification(request: request)
    }
}